### PR TITLE
Smartbanner for dynamic web app (vuejs)

### PR DESCRIPTION
**Pull request by Beauceron**
This update is for adding the possibility to recall the parsing when need with Single Page Application when the data come from a Database. Please provide comments if need I will make changes if needed.

### CODE REVIEW

**Review by ain — CHANGES_REQUESTED** [review]
Thanks for the great contribution @Beauceron!

With Code Review completed, I'd call you to also add the specs in areas affected. 

I added Coveralls support in [aee68f5](https://github.com/ain/smartbanner.js/commit/aee68f5dac61959d8864351aaba68b988625dacf) now so you'd be able to get a report on your next commit and an overview on your changes at https://coveralls.io/github/ain/smartbanner.js

Let me know if you have any questions/issues!

**Comment by Beauceron** [comment]
@ain I also have the eslint failing because of this rule : 
```
    "linebreak-style": [
      "error",
      "unix"
    ],
```
I have all the JS files in CRLF on local but eslint recommand LF. 

Is this my IDE that make that or you have the same thing?

File: src/smartbanner.js [hunk 1]
```diff
@@ -210,7 +215,9 @@ export default class SmartBanner {
     }
 
     let bannerDiv = document.createElement('div');
-    document.querySelector('body').appendChild(bannerDiv);
+    const body = document.querySelector('body');
+    body.appendChild(bannerDiv);
+    body.classList.add(OPEN_BODY_CLASS);
```
[the remark this answers is not in the document]
Comment (Beauceron) [reply src/smartbanner.js:220]: @ain I added some test please take a look.

**Comment by ain** [comment]
@Beauceron no, I don't have that problem as Vim is Unix-compatible by default.

In most IDEs you are able to save files in Unix format as opposed to Windows.

The reasoning of the LF requirement is that the code is mostly run on the Unix-compatible platforms, e.g. Linux.

Could you please merge the latest `main` branch into your PR and retry linting?

**Comment by ain** [comment]
We had to withdraw the change in `main` as per the errors seen, other solutions need to be found for `eslint` failures.

**Pull request closed without merging**